File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    const workDays = progresses.map((el,i)=>Math.ceil((100-el) / speeds[i]));
    console.log("workDays : ", workDays);
    let prevMaxDay = workDays[0];
    let deployNum = 0;
    for(let i=0; i<workDays.length; ++i){
        console.log("deployNum : ", deployNum, " prevMaxDay : " , prevMaxDay, " workDays[i] : " , workDays[i])
        if(prevMaxDay >= workDays[i]){
            ++deployNum;
        }else{
            console.log("anwerPush!! deployNum : ", deployNum);
            answer.push(deployNum);
            deployNum = 1;
            prevMaxDay = workDays[i];
        }
    }
    answer.push(deployNum);
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cs
using System;
using System.Collections.Generic;
using System.Linq;
public class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<int> answer = new List<int>();
        /*List<int> workDays = new List<int>();
        for(int i=0; i<progresses.Length; ++i)
        {
            workDays.Add((int)Math.Ceiling((100f - progresses[i]) / speeds[i]));
        }
        */
        List<int> workDays = progresses.Select((value, i)=>(int)Math.Ceiling((100f- value)/speeds[i])).ToList();
        
        int prevMaxDeployDay = workDays[0];
        int deployNumOfDay = 0;
        for(int i=0; i<workDays.Count; ++i)
        {
            if(prevMaxDeployDay >= workDays[i])
            {
                ++deployNumOfDay;
            }
            else
            {
                answer.Add(deployNumOfDay);
                deployNumOfDay = 1;
                prevMaxDeployDay = workDays[i];
            }
        }
        answer.Add(deployNumOfDay);
        return answer.ToArray();
    }
}